rust/relay/ebpf-turn-router/src/move_payload.rs

```diff
@@ -0,0 +1,149 @@
+use crate::{channel_data::CdHdr, error::Error};
+use aya_ebpf::{helpers::bpf_xdp_adjust_tail, programs::XdpContext};
+use network_types::{
+    eth::EthHdr,
+    ip::{Ipv4Hdr, Ipv6Hdr},
+    udp::UdpHdr,
+};
+
+// Set an upper limit for bounds checks
+const MAX_MTU: usize = 1500;
+const MAX_PAYLOAD: usize = MAX_MTU - EthHdr::LEN - Ipv4Hdr::LEN - UdpHdr::LEN - CdHdr::LEN;
+
+#[inline(always)]
+pub fn add_channel_data_header_ipv4(ctx: &XdpContext, header: CdHdr) -> Result<(), Error> {
+    adjust_header::<{ Ipv4Hdr::LEN }>(ctx, Some(&header))
+}
+
+#[inline(always)]
+pub fn add_channel_data_header_ipv6(ctx: &XdpContext, header: CdHdr) -> Result<(), Error> {
+    adjust_header::<{ Ipv6Hdr::LEN }>(ctx, Some(&header))
+}
+
+#[inline(always)]
+pub fn remove_channel_data_header_ipv4(ctx: &XdpContext) -> Result<(), Error> {
+    adjust_header::<{ Ipv4Hdr::LEN }>(ctx, None)
+}
+
+#[inline(always)]
+pub fn remove_channel_data_header_ipv6(ctx: &XdpContext) -> Result<(), Error> {
+    adjust_header::<{ Ipv6Hdr::LEN }>(ctx, None)
+}
+
+/// Add or remove channel data header
+#[inline(never)]
+fn adjust_header<const IP_HEADER_LEN: usize>(
+    ctx: &XdpContext,
+    header: Option<&CdHdr>,
+) -> Result<(), Error> {
+    let offset = EthHdr::LEN + IP_HEADER_LEN + UdpHdr::LEN;
+
+    if let Some(hdr) = header {
```

Comment:
Similar here. There is basically no shared functionality between these two branches.

I think it is cleaner to have two functions here.